ste/sender-blockBlobFromLocal.go

```diff
@@ -86,26 +83,18 @@ func (u *blockBlobUploader) generatePutWholeBlob(id common.ChunkID, blockIndex i
 
 	return createSendToRemoteChunkFunc(u.jptm, id, func() {
 		jptm := u.jptm
+		encodedBlockID := u.generateEncodedBlockID()
+		u.setBlockID(blockIndex, encodedBlockID)
 
 		// Upload the blob
 		jptm.LogChunkStatus(id, common.EWaitReason.Body())
 		var err error
 		if jptm.Info().SourceSize == 0 {
-			_, err = u.destBlockBlobURL.Upload(jptm.Context(), bytes.NewReader(nil), u.headersToApply, u.metadataToApply, azblob.BlobAccessConditions{})
+			u.blockIDs = make([]string, 0) // Empty block ID list
 		} else {
-			// File with content
-
-			// Get the MD5 that was computed as we read the file
-			md5Hash, ok := <-u.md5Channel
-			if !ok {
-				jptm.FailActiveUpload("Getting hash", errNoHash)
-				return
-			}
-			u.headersToApply.ContentMD5 = md5Hash
-
 			// Upload the file
 			body := newPacedRequestBody(jptm.Context(), reader, u.pacer)
-			_, err = u.destBlockBlobURL.Upload(jptm.Context(), body, u.headersToApply, u.metadataToApply, azblob.BlobAccessConditions{})
+			_, err = u.destBlockBlobURL.StageBlock(jptm.Context(), encodedBlockID, body, azblob.LeaseAccessConditions{}, nil)
 		}
```

Comment:
I don't understand this change. Using Stage Block -> Commit block list will destroy performance with small file uploads.

Reply:
I'm... actually not entirely sure what was running through my head when I did that.

Going to undo that change and just have the block blob cleanup behavior altered.